server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -908,7 +1009,7 @@ private boolean segmentExists(final Handle handle, final DataSegment segment)
    * Read dataSource metadata. Returns null if there is no metadata.
    */
   @Override
-  public DataSourceMetadata getDataSourceMetadata(final String dataSource)
+  public @Nullable DataSourceMetadata retrieveDataSourceMetadata(final String dataSource)
```

Comment:
I prefer putting `@Nullable` before the type because semantically it characterizes the returned type, not the method (there are also modifiers and other annotations in between).